core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -0,0 +1,167 @@
+      // the initialization is successful (judged by the success of
+      // initializing data.questions), and false otherwise.
+      tryToInit: function() {
+        data.title = SimpleEditorShimService.getTitle();
+        data.introductionHtml = SimpleEditorShimService.getIntroductionHtml();
+        data.questions = StatesToQuestionsService.getQuestions();
+        return Boolean(data.questions);
+      },
+      getData: function() {
+        return data;
+      },
+      getTitle: function() {
+        return data.title;
+      },
+      getIntroductionHtml: function() {
+        return data.introductionHtml;
+      },
+      getQuestions: function() {
+        return data.questions;
+      },
+      saveTitle: function(newTitle) {
+        SimpleEditorShimService.saveTitle(newTitle);
+        data.title = newTitle;
+      },
+      saveIntroductionHtml: function(newIntroductionHtml) {
+        SimpleEditorShimService.saveIntroductionHtml(newIntroductionHtml);
+        data.introductionHtml = newIntroductionHtml;
+      },
+      saveCustomizationArgs: function(stateName, newCustomizationArgs) {
+        SimpleEditorShimService.saveCustomizationArgs(
+          stateName, newCustomizationArgs);
+        getBindableQuestion(stateName).setInteractionCustomizationArgs(
+          newCustomizationArgs);
+      },
+      saveAnswerGroups: function(stateName, newAnswerGroups) {
+        SimpleEditorShimService.saveAnswerGroups(stateName, newAnswerGroups);
+        getBindableQuestion(stateName).setAnswerGroups(newAnswerGroups);
+      },
+      saveDefaultOutcome: function(stateName, newDefaultOutcome) {
+        SimpleEditorShimService.saveDefaultOutcome(
+          stateName, newDefaultOutcome);
+        getBindableQuestion(stateName).setDefaultOutcome(newDefaultOutcome);
+      },
+      saveBridgeHtml: function(stateName, newHtml) {
+        // This is actually the content HTML for the *next* state.
+        SimpleEditorShimService.saveBridgeHtml(
+          getNextStateName(stateName), newHtml);
+        getBindableQuestion(stateName).setBridgeHtml(newHtml);
+      },
+      addNewQuestion: function() {
+        // This effectively adds a new multiple-choice interaction to the
+        // latest state in the chain.
+        var lastStateName = (
+          data.questions.length > 0 ?
+          data.questions[data.questions.length - 1].getDestinationStateName() :
+          SimpleEditorShimService.getInitStateName());
+
+        SimpleEditorShimService.saveInteractionId(
+          lastStateName, DEFAULT_INTERACTION.ID);
+        SimpleEditorShimService.saveCustomizationArgs(
+          lastStateName, DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+
+        var stateData = SimpleEditorShimService.getState(lastStateName);
+        data.questions.push(QuestionObjectFactory.create(
+          lastStateName, stateData.interaction, ''));
+      },
+      canAddNewQuestion: function() {
+        if (!data.questions) {
```

Comment:
It is needed, but that's probably because I'm using ng-show. I'll change it to ng-if and see if that works better, otherwise I might need to keep this.